app/src/main/java/io/homeassistant/companion/android/widgets/WidgetDynamicFieldAdapter.kt

```diff
@@ -127,7 +128,11 @@ class WidgetDynamicFieldAdapter(
         // Currently value can by Any? but will currently only be storing String?
         // This may have to be changed later if multi-select gets implemented
         if (serviceFieldList[position].value != null) {
-            serviceFieldList[position].value as String
+            try {
+                serviceFieldList[position].value as String
+            } catch (e: Exception) {
+                Log.d("WidgetField", "Unable to get service field list", e)
```

Comment:
We should extract the TAG to a variable.
```suggestion
                Log.d(TAG, "Unable to get service field list", e)
```